bitcoin/src/blockdata/transaction.rs

```diff
@@ -71,10 +76,15 @@ impl Wtxid {
     /// witness commitment tree) since the coinbase transaction contains a commitment to all
     /// transactions' wTXIDs but naturally cannot commit to its own.
     pub const COINBASE: Self = Self::from_byte_array([0; 32]);
+}
 
-    /// The "all zeros" wTXID.
-    #[deprecated(since = "TBD", note = "use Wtxid::COINBASE instead")]
-    pub fn all_zeros() -> Self { Self::COINBASE }
+crate::internal_macros::define_extension_trait! {
+    /// Extension functionality for the [`Wtxid`] type.
+    pub trait WtxidExt impl for Wtxid {
+        /// The "all zeros" wTXID.
+        #[deprecated(since = "TBD", note = "use Wtxid::COINBASE instead")]
+        fn all_zeros() -> Self { Self::COINBASE }
+    }
```

Comment:
I wonder if it'd be better to have only one trait for both.

Reply:
If it is, can we do it as a follow up? (I don't personally think it is though because these traits are super uninteresting and should just follow the naming scheme `FooExt` IMO.

Reply:
I was thinking of something like `HashExt`. The reason is I was worried about there being conflicting possibilities for the method but maybe it's fine when none of them is implemented for both. If you can write some quick test (note: you need to actually do three crates to accurately simulate the situation), that'd be nice, I'll need to run soon.

Reply:
If we want to be sure of non-conflicting methods, then using `TypeExt` which is *only* implemented for `Type` is the way to go. As soon as we start trying to share extension traits, that's when we run into concerns about overlap.

Also, in general I would advise against trying to write traits that are generic over TXID and wTXID in user-facing code. Mainly because such traits are practically impossible to find good names for.

Reply:
Lets keep it simple and braindead, trait `FooExt` extends `Foo` and that's all. Then folk can guess the extension names without having to look them up.

Reply:
OK, makes sense.